jira-config/jira-config-ui/src/main/resources/JIRA/JIRAAuth/BasicAuthConfigDisplayer.xml

```diff
@@ -0,0 +1,234 @@
+
+<!--
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+
+<xwikidoc version="1.5" reference="JIRA.JIRAAuth.BasicAuthConfigDisplayer" locale="">
+  <web>JIRA.JIRAAuth</web>
+  <name>BasicAuthConfigDisplayer</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title/>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity}}
+#set ($basicAuthClass = 'JIRA.JIRAAuth.BasicAuthClass')
+#set ($confDoc = $xwiki.getDocument('JIRA.JIRAAuth.BasicAuthConfiguration'))
+#set ($confObjects = $confDoc.getObjects($basicAuthClass))
+#if ($xcontext.action == 'get')
+  #if ($request.getParameter('save') == 'true')
+    ## Save the changes
+    #if ($services.csrf.isTokenValid($!request.form_token))
+      #foreach ($obj in $confObjects)
+        #set ($formIdFieldName = "${basicAuthClass}_${obj.number}_id")
+        #set ($formUsernameFieldName = "${basicAuthClass}_${obj.number}_username")
+        #set ($formPwdFieldName = "${basicAuthClass}_${obj.number}_password")
+        #set ($formIdValue = $request.getParameter($formIdFieldName))
+        #set ($formUsernameValue = $request.getParameter($formUsernameFieldName))
+        #set ($formPwdValue = $request.getParameter($formPwdFieldName))
+        #if ($formIdValue != $NULL &amp;&amp; $formUsernameValue != $NULL &amp;&amp; $formPwdValue != $NULL)
+          #set ($discard = $obj.set('id', $formIdValue))
+          #set ($discard = $obj.set('username', $formUsernameValue))
+          #set ($discard = $obj.set('password', $formPwdValue))
+        #else
+          ## Seem that the object is not any more in the form, so drop it
+          #set ($discard = $confDoc.removeObject($obj))
+        #end
+      #end
+      #set ($discard = $confDoc.save())
+    #else
+      #set($discard = $response.setStatus(403))
+      #jsonResponse({"error": "Invalid CSRF token"})
```

Comment:
I'm having a doubt: isn't `#jsonResponse` always sending a 200 status code?

Reply:
Checked the code and tested an it seem working 🙂